build.gradle

```diff
@@ -719,22 +719,34 @@ defineProperty("TEST_ONLY", "false")
 ext.IS_TEST_ONLY = Boolean.parseBoolean(TEST_ONLY) || !buildSdkForTest
 
 ext.IS_TEST_SDK = false
+ext.TEST_SDK_DIR = ""
+
 if (hasProperty("TEST_SDK_PATH")) {
     if (!IS_TEST_ONLY) {
         fail("TEST_ONLY must be set to true when using an external TEST_SDK_PATH for running tests: include -PTEST_ONLY=true")
     }
-    def sdkShimsPath = file("${TEST_SDK_PATH}")
-    def sdkPath = file("${TEST_SDK_PATH}/sdk")
-    def shimsPath = file("${TEST_SDK_PATH}/shims")
-    if (!sdkShimsPath.exists() || !sdkShimsPath.isDirectory() ||
-             !sdkPath.exists() || !sdkPath.isDirectory() ||
-           !shimsPath.exists() || !shimsPath.isDirectory()) {
-        fail("The provided TEST_SDK_PATH=${TEST_SDK_PATH} is invalid")
+    String testSdkPath = "${TEST_SDK_PATH}"
+    if (testSdkPath.startsWith("~")) {
```

Comment:
This isn't related to handling relative paths at all. Rather it is related to how bash works. The `~` character is handled by the shell and it is a known limitation that you can't use it in the middle of a string.

Reply:
Thanks Kevin, I shall remove the part related to `~`